scan-command/src/main/java/io/ballerina/scan/internal/ProjectAnalyzer.java

```diff
@@ -51,4 +65,51 @@ private Consumer<DocumentId> analyzeDocument(Module module, ScannerContextImpl s
             analyzer.analyze();
         };
     }
+
+    List<Issue> runExternalAnalyzers(Project project) {
+        StringBuilder newImports = new StringBuilder();
+        StringBuilder tomlDependencies = new StringBuilder();
+        int importCounter = 0;
+
+        for (ScanTomlFile.Analyzer analyzer : scanTomlFile.getAnalyzers()) {
+            String reportingSource = analyzer.getOrg() + FORWARD_SLASH + analyzer.getName();
+            String analyzerImport = IMPORT_PREFIX + reportingSource + USE_IMPORT_AS_SERVICE;
+            newImports.append(analyzerImport).append("\n");
+
+            if (analyzer.getVersion() != null) {
+                tomlDependencies.append("\n");
+                tomlDependencies.append("[[dependency]]" + "\n");
+                tomlDependencies.append("org='").append(analyzer.getOrg()).append("'\n");
+                tomlDependencies.append("name='").append(analyzer.getName()).append("'\n");
+                tomlDependencies.append("version='").append(analyzer.getVersion()).append("'\n");
+
+                if (analyzer.getRepository() != null) {
+                    tomlDependencies.append("repository='").append(analyzer.getRepository()).append("'\n");
+                }
+            }
+            importCounter++;
+        }
+
+        Module defaultModule = project.currentPackage().getDefaultModule();
+        Document mainBAL = defaultModule.document(defaultModule.documentIds().iterator().next());
+        String documentContent = mainBAL.textDocument().toString();
+        mainBAL.modify().withContent(newImports + documentContent).apply();
+
+        BallerinaToml ballerinaToml = project.currentPackage().ballerinaToml().orElse(null);
+        if (ballerinaToml != null) {
+            documentContent = ballerinaToml.tomlDocument().textDocument().toString();
+            ballerinaToml.modify().withContent(documentContent + tomlDependencies).apply();
+        }
+
+        List<ScannerContext> scannerContextList = new ArrayList<>();
+
+        // TODO: Implement the logic to pass scanner contexts to compiler plugins and perform external static analysis.
+
+        List<Issue> externalIssues = new ArrayList<>();
+        for (ScannerContext scannerContext : scannerContextList) {
+            ReporterImpl reporter = (ReporterImpl) scannerContext.getReporter();
+            externalIssues.addAll(reporter.getIssues());
+        }
```

Comment:
Do you need a separate list? Why not populate `externalIssues` directly where you will be (instead of) adding to `scannerContextList`?

Reply:
This issue will be addressed in the next PR